tests/e2e/kafka_cluster_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/banzaicloud/koperator/api/v1beta1"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+const (
+	kafkaKind        = "kafkaclusters.kafka.banzaicloud.io"
+	kafkaClusterName = "kafka"
+)
+
+// createKafkaClusterIfDoesNotExist creates a kafka cluster if
+// there isn't a preexisting one
+func createKafkaClusterIfDoesNotExist(kubectlOptions *k8s.KubectlOptions, path string) {
+	By("Checking existing kafkaClusters")
+	err := checkExistenceOfK8sResource(kubectlOptions, kafkaKind, kafkaClusterName)
+
+	if err == nil {
+		fmt.Printf("Kafka cluster %s already exists", kafkaClusterName)
+	} else {
+		By("Deploying a KafkaCluster")
+		createK8sResourcesFromManifest(kubectlOptions, path, false)
```

Comment:
Here we should use sophisticated resource deployment to make version compatible the simplekafkacluster.
Because our koperator deployment test can run in different koperator versions (https://github.com/banzaicloud/koperator/blob/d9b806a3008a1fee2c30fc26eb3535a70e7ce541/tests/e2e/koperator_test.go#L124)
It would be good to make capable to use the corresponding kafkacluster sample file.
I wrote a method for this usecase what can be used: https://github.com/banzaicloud/koperator/blob/d9b806a3008a1fee2c30fc26eb3535a70e7ce541/tests/e2e/koperator_test.go#L71